cross_list_courses/static/cross_list_courses/js/controllers/ListController.js

```diff
@@ -3,103 +3,272 @@
      * Angular controller for the cross listing page.
      */
     var app = angular.module('CrossListCourses');
-    app.controller('ListController',
-        ['$scope', '$http', '$timeout', '$document', '$window', '$compile','djangoUrl','$log',
-        function($scope, $http, $timeout, $document, $window, $compile, $djangoUrl, $log){
+    app.controller('ListController', ListController);
 
-            $scope.dtInstance = null;
+    function ListController($scope, $http, $timeout, $document, $window,
+                            $compile, djangoUrl, $log, $q, $uibModal) {
+        // {alertType: 'bootstrap alert class', text: 'actual message'}
```

Comment:
That's meant to indicate the expected `$scope.message` format -- I can update the code to make that clearer.